hera_pspec/tests/test_grouping.py

```diff
@@ -76,6 +76,54 @@ def test_group_baselines(self):
         for i in range(len(g1)):
             for j in range(len(g1[i])):
                 self.assertEqual(g1[i][j], g3[i][j])
+
+    def test_average_spectra(self):
+        """
+        Test average spectra behavior.
+        """
+        dfile = os.path.join(DATA_PATH, 'zen.all.xx.LST.1.06964.uvA')
+        # Load into UVData objects
+        uvd = UVData()
+        uvd.read_miriad(dfile)
+        cosmo = conversions.Cosmo_Conversions()
+        beamfile = os.path.join(DATA_PATH, 'HERA_NF_dipole_power.beamfits')
+        uvb = pspecbeam.PSpecBeamUV(beamfile, cosmo=cosmo)
+        # find conversion factor from Jy to mK
+        Jy_to_mK = uvb.Jy_to_mK(np.unique(uvd.freq_array), pol='XX')
+
+        # reshape to appropriately match a UVData.data_array object and multiply in!
+        uvd.data_array *= Jy_to_mK[None, None, :, None]
+        # slide the time axis of uvd by one integration
+        uvd1 = uvd.select(times=np.unique(uvd.time_array)[:-1:2], inplace=False)
+        uvd2 = uvd.select(times=np.unique(uvd.time_array)[1::2], inplace=False)
+
+        # Create a new PSpecData object, and don't forget to feed the beam object
+        ds = pspecdata.PSpecData(dsets=[uvd1, uvd2], wgts=[None, None], beam=uvb)
+        ds.rephase_to_dset(0)
+        # change units of UVData objects
+        ds.dsets[0].vis_units = 'mK'
+        ds.dsets[1].vis_units = 'mK'
+        
+        baselines = [(24,25), (37,38), (38,39)]
+        # calculate all baseline pairs from group
+        baselines1, baselines2, blpairs = utils.construct_blpairs(baselines, exclude_auto_bls=True, 
+                                                                 exclude_permutations=True)
+        uvp = ds.pspec(baselines1, baselines2, (0, 1), [('xx', 'xx')], spw_ranges=[(300, 400)], input_data_weight='identity',
+               norm='I', taper='blackman-harris', verbose=False)
+        keys = uvp.get_all_keys()
+        Pn = uvp.generate_noise_spectra(0, 'xx', 220)
+        for key in keys:
+            blp = uvp.antnums_to_blpair(key[1])
+            error = Pn[blp]
+            uvp.set_stats("noise", key, error)
+        blpair_groups = [[((24, 25), (37, 38)),((24, 25), (38, 39)), ((37, 38), (38, 39))]]
+        uvp_avg_ints_wgts = grouping.average_spectra(uvp, blpair_groups=blpair_groups,
+                                                error_field="noise", time_avg=True,inplace=False)
+        uvp_avg_error_wgts = grouping.average_spectra(uvp, time_avg=True, blpair_groups=blpair_groups,
+                                                 error_field="noise", error_weights="noise",
+                                                 inplace=False)
+        assert(abs(uvp_avg_ints_wgts.stats_array["noise"][0][0,0,0]) < abs(uvp.stats_array["noise"][0][0,0,0]))
```

Comment:
can you add a comment here on what this is testing and to what tolerance?

Reply:
The intuition here is the error bar on the average result should be smaller than on one sample.

Reply:
I see, given that we have known input, can we assert that the result is equal to an analytic result? In this case, we are averaging 3 blpairs, so the averaged noise should be 1/sqrt(N) times the single sample errorbar, no?

Reply:
Since the noise power spectrum for each single baseline-pair power spectrum are not equal,  the average result will not be the exactly 1/sqrt(N) times the single sample errorbar. If we want to be confirmed on this, I did a similar thing in  test_stats_array, where all the error bars are set to be one, and I have a check like  nt.assert_true(np.all(np.isclose(u.get_stats("errors", keys[0])[0], np.ones(u.Ndlys)/np.sqrt(len(blpairs))))).

Reply:
let's add that. once that's in I'm all set with this PR!